src/app/agent/page.tsx

```diff
@@ -270,133 +283,134 @@ const AgentPageComponent: React.FC = () => {
+    setLlmTokenLatency(0);
+    setTtsLatency(0);
+    if (voiceSession) {
+      console.log(`handleStart - starting voice session ${voiceSession}`);
+      voiceSession!.start();
+    }
+  }, [voiceSession]);
+
+  // Create the voice session, configure it, and warm it up.
+  const createSession = useCallback(() => {
+    const fixieClient = new FixieClient({});
+    const voiceInit: VoiceSessionInit = {
+      webrtcUrl: webrtcUrl || "wss://wsapi.fixie.ai",
+      asrProvider: asrProvider,
+      ttsProvider: ttsProvider,
+      ttsVoice: ttsVoice,
+      model: model,
+    };
+    const session = fixieClient.createVoiceSession({
       agentId,
-      docs,
-      webrtcUrl,
+      init: voiceInit,
     });
-    setChatManager(manager);
-    manager.onStateChange = (state) => {
-      switch (state) {
-        case ChatManagerState.LISTENING:
-          setHelpText("Listening...");
-          break;
-        case ChatManagerState.THINKING:
-          setHelpText(`Thinking... ${tapOrClick.toLowerCase()} to cancel`);
-          break;
-        case ChatManagerState.SPEAKING:
-          setHelpText(`Speaking... ${tapOrClick.toLowerCase()} to interrupt`);
-          break;
-        default:
-          setHelpText(idleText);
-      }
+
+    session.onStateChange = (state: VoiceSessionState) => {
+      setState(state);
     };
-    manager.onInputChange = (text, final) => {
+    session.onInputChange = (text: string, final: boolean) => {
       setInput(text);
     };
-    manager.onOutputChange = (text, final) => {
+    session.onOutputChange = (text: string, final: boolean) => {
       setOutput(text);
       if (final) {
         setInput("");
       }
     };
-    manager.onLatencyChange = (kind, latency) => {
-      switch (kind) {
+    session.onLatencyChange = (metric: string, value: number) => {
+      switch (metric) {
         case "asr":
-          setAsrLatency(latency);
+          setAsrLatency(value);
           setLlmResponseLatency(0);
           setLlmTokenLatency(0);
           setTtsLatency(0);
           break;
         case "llm":
-          setLlmResponseLatency(latency);
+          setLlmResponseLatency(value);
           break;
         case "llmt":
-          setLlmTokenLatency(latency);
+          setLlmTokenLatency(value);
           break;
         case "tts":
-          setTtsLatency(latency);
+          setTtsLatency(value);
           break;
       }
     };
-    manager.onError = () => {
-      manager.stop();
+    session.onError = () => {
+      session.stop();
     };
-    return () => manager.stop();
-  };
-  const changeAgent = (delta: number) => {
+    console.log(`init - warming up voice session ${session}`);
+    session.warmup();
+    setVoiceSession(session);
+    return session;
+  }, [agentId, asrProvider, model, ttsProvider, ttsVoice, webrtcUrl]);
+
+  // Called when the main button is clicked.
+  const onButtonClick = useCallback(() => {
+    if (!started) {
+      const session = createSession();
```

Comment:
the way the session is passed around here seems kind of weird (ie not as an argument to handleStart), and it seems the session is stared in multiple places. I also think we want to create and warm up the session on pageload.

Reply:
So, I'm focused right now on *only* getting the app to work with the Voice SDK, but I realize there are a few ways the initialization logic and such could be improved. You'll recall I spent hours beating my head against this in the HiSanta app -- so, I'm just doing something a little crude that works for now. What I ended up doing in HiSanta was more complex (in part because of the ringtone support, but also because of the same stuff you're noticing here), and in keeping with "lift and shift" as much as possible I didn't want to go rewriting this component from scratch to make it work the same way as HiSanta.

I'm happy to go do that and clean all this stuff up, however, it's gonna take some time and I was under the impression we just wanted a quick-and-dirty port so we could move onto the Voice SDK as a starting point.

I agree that we want to be able to create and warm up the session on pageload. Unfortunately that is hard to do with the way React useEffect works, and will require more rework. Happy to do that in follow PRs.

Reply:
The thing is that the existing code does do the pageload warmup, and seems to work fine. If we can't preserve that in the updated code, can we just lift-and-shift the existing HTML/JS code then?